### PR TITLE
Closes #261: Modernise VFQ template and vignette

### LINKED ISSUE #261: Modernise VFQ test data, template and vignete
### Background Information

The ADVFQ template, and also the test data, are quite old. We could do with a review to see if anything can be modernised/updated. At Roche, we have our own version of ADVFQ which was originally based on this template, so we could look to harmonise the two.

### Definition of Done

- [ ] Perform review as described above.
- [ ] Make updates to template, vignette and data if needed. Can also split into separate issues (also because test data now lives in pharmaversesdtm).

### PR DESCRIPTION
**Pull request by manciniedoardo**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update).
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files.
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions]
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately.
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralophtha (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralophtha site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralophtha/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`.
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`.
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by manciniedoardo — COMMENTED** [review]
Thanks @bundfussr for the helpful review - i believe i have actioned all comments

File: inst/templates/ad_advfq.R [hunk 1]
```diff
@@ -21,239 +21,190 @@ qs <- qs_ophtha
...
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = str_remove(QSTEST, "VFQ1-"),
+    PARCAT1 = "VFQ-25 INTERVIEWER ADMINISTERED",
+    PARCAT2 = "Original Items"
+  )
 
-adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01A, TRT01P)
+## Transformed parameters ----
+param_lookup_transformed <- tribble(
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/templates/ad_advfq.R:82]: will keep as is for simplicity

File: vignettes/advfq.Rmd [hunk 2]
```diff
@@ -39,192 +46,557 @@ library(admiral)
...
+advfq <- advfq %>%
+  call_derivation(
+    derivation = derive_extreme_records,
+    dataset = .,
+    dataset_add = .,
     by_vars = c(
       get_admiral_option("subject_keys"),
-      exprs(!!!adsl_vars, PARAMCD, VISITNUM, VISIT)
+      exprs(PARAMCD, AVISIT, AVISITN, ADT, ADY)
     ),
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply vignettes/advfq.Rmd]: agree, i'll move what i need to the keep_source_vars (otherwise vars like USUBJID are not populated for the new records) and remove the by vars.

File: inst/templates/ad_advfq.R [hunk 3]
```diff
@@ -21,290 +21,421 @@ qs <- qs_ophtha
...
-  "QSG02", "QSG02", "General Score 02", "NEI VFQ-25", "Averaged Result",
-  "QBCSCORE", "QBCSCORE", "Composite Score", "NEI VFQ-25", "Averaged Result"
-)
-attr(param_lookup$QSTESTCD, "label") <- "Question Short Name"
+# Lookup tables ----
+
+# nolint start
+
+## Original parameters (note PARAM, PARAMCD are identical to QSTEST, QSTESTCD) ----
+param_lookup_original <- tribble(
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/templates/ad_advfq.R]: not sure, will fix in both template and vignette

File: inst/templates/ad_advfq.R [hunk 4]
```diff
@@ -21,290 +21,421 @@ qs <- qs_ophtha
...
+  "VFQ1A08", "Difficulty Seeing Programs on TV", "distance vision", "Distance Activities", "Optional Item",
+  "VFQ1A09", "Difficulty Entertaining Friends", "social fx", "Vision Specific: Social Functioning", "Optional Item",
+  "VFQ1A11A", "Do You Have More Help From Others", "role limitations", "Vision Specific: Role Difficulties", "Optional Item",
+  "VFQ1A11B", "Limited in Kinds of Things Can Do", "role limitations", "Vision Specific: Role Difficulties", "Optional Item",
+  "VFQ1A12", "Often Irritable Because Eyesight", "well-being/distress", "Vision Specific: Mental Health", "Optional Item",
+  "VFQ1A13", "I Don't Go Out of My Home Alone", "dependency", "Vision Specific: Dependency", "Optional Item"
+) %>%
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = str_remove(QSTEST, "VFQ1-"),
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/templates/ad_advfq.R]: will do, thanks

**Review by bundfussr — APPROVED** [review]
